airflow/dag_processing/processor.py

```diff
@@ -745,25 +746,33 @@ def execute_callbacks_without_dag(
     @provide_session
     def _execute_dag_callbacks(self, dagbag: DagBag, request: DagCallbackRequest, session: Session):
         dag = dagbag.dags[request.dag_id]
-        dag_run = dag.get_dagrun(run_id=request.run_id, session=session)
-        dag.handle_callback(
-            dagrun=dag_run, success=not request.is_failure_callback, reason=request.msg, session=session
+        dagrun = DAG.fetch_dagrun(dag_id=dag.dag_id, run_id=request.run_id, session=session)
+        callbacks, context = (
+            DAG.fetch_callback(
+                dag=dag,
+                dagrun=dagrun,
+                success=not request.is_failure_callback,
+                reason=request.msg,
+                session=session,
+            )
+            or None,
+            None,
         )
```

Comment:
This looks wrong? If `fetch_callback` returns something, it is alreay a tuple and this would incorrectly result in a nested structure